### PR TITLE
Fix artifact file name and support PNG

### PR DESCRIPTION
**Pull request by akiakaba**
Some improvements

- Support PNG
- Fix artifact file name (duplicated hyphens and underscores) **BRAKING CHANGE**
- Some refactoring
  - Extract ShieldIO module

### CODE REVIEW

File: src/shields-io.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+'use strict'
+
+const apiRoot = 'https://img.shields.io/badge/';
+const convert4shieldsIo = str => str.replace(/-/g, '--').replace(/_/g, '__');
+const toUrl = (subject, coverage, color, imageType) => `${apiRoot}${subject}-${coverage}%25-${color}.${imageType}`;
+
+module.exports = {
+  url(param) {
+    const subject = convert4shieldsIo(param.subject);
+    const coverage = convert4shieldsIo(param.coverage);
```
Comment (sisisin) [inline src/shields-io.js]: `convert4shieldsIo` needs string parameter but `param.coverage` is number.
So this line rise `Type Error`, please fix this.

Comment (LagunaPresa) [inline src/shields-io.js]: thx!

Comment (akiakaba) [reply src/shields-io.js]: thx!

**Comment by sisisin** [comment]
LGTM, thanks.

**Pull request merged**